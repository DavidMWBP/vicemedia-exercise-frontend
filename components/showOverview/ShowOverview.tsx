import Show, { Show as ShowInterface } from "components/showOverview/show/Show";
import {
	StyledShowOverview,
	StyledShowSelectorWrapper
} from "components/showOverview/ShowOverview.styled";
import ShowSelector from "components/showOverview/showSelector/ShowSelector";
import { AnimatePresence, motion } from "framer-motion";
import findShow from "lib/findShow";
import useShows from "lib/hooks/useShows";
import { useRouter } from "next/dist/client/router";
import { useEffect, useState } from "react";

export type Shows = Array<ShowInterface>;
export type ShowId = string | string[] | undefined;

export interface ShowOverviewProps {
	/** Shows fetched from an endpoint to render */
	shows: Shows;
	/** show id fetched from query param on initial load */
	showId?: ShowId;
}

function ShowOverview({
	shows,
	showId
}: ShowOverviewProps): JSX.Element | null {
	const { query } = useRouter();
	const { data, isError } = useShows(shows);
	const [selectedShow, setSelectedShow] = useState<ShowInterface | null>(
		findShow(showId, shows)
	);

	useEffect(() => {
		setSelectedShow(findShow(query.id, data));
	}, [query.id, data]);

	if (isError || !data) return <div>No shows have been found!</div>;

	return (
		<StyledShowOverview>
			<AnimatePresence>
				<div>
					{selectedShow && (
						<motion.div
							key={selectedShow.id}
							initial={{ scale: 0 }}
							animate={{ scale: 1 }}
							transition={{ type: "spring", bounce: 0.25, duration: 0.25 }}
							exit={{ scale: 0 }}
						>
							<Show {...selectedShow} />
						</motion.div>
					)}
				</div>
			</AnimatePresence>
			<StyledShowSelectorWrapper>
				{data?.map((show, index) => {
					return (
						<ShowSelector
							key={`show_${show.id}_${index}`}
							{...show}
							index={index}
							isActive={selectedShow?.id === show.id}
						/>
					);
				})}
			</StyledShowSelectorWrapper>
		</StyledShowOverview>
	);
}

export default ShowOverview;
