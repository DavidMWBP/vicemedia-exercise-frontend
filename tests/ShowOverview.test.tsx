import ShowOverview from "components/showOverview/ShowOverview";
import React from "react";
import renderer from "react-test-renderer";
import showsData from "shows.json";

jest.mock("next/dist/client/router", () => ({
	__esModule: true,
	useRouter: () => ({
		query: {},
		push: jest.fn(() => Promise.resolve(true))
	})
}));

it("renders correctly with the show titled 'Love Industries'", () => {
	const tree = renderer
		.create(<ShowOverview shows={showsData.shows} showId="e6" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
