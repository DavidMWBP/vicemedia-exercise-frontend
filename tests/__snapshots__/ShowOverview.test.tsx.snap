// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with the show titled 'Love Industries' 1`] = `
<div
  className="ShowOverviewstyled__StyledShowOverview-xnlc2h-0 jucnhV"
>
  <div>
    <div
      style={
        Object {
          "transform": "scale(0) translateZ(0)",
        }
      }
    >
      <div
        className="Showstyled__StyledShow-sc-7uvw6o-0 cGpFxR"
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "inline-block",
              "margin": 0,
              "maxWidth": "100%",
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "display": "block",
                "maxWidth": "100%",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjkwIiBoZWlnaHQ9IjQzMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style={
                Object {
                  "border": "none",
                  "display": "block",
                  "margin": 0,
                  "maxWidth": "100%",
                  "padding": 0,
                }
              }
            />
          </div>
          <img
            alt="Love Industries"
            className="Showstyled__StyledImage-sc-7uvw6o-1"
            decoding="async"
            src="/_next/image?url=%2Fimages%2Floveindustries.jpg&w=640&q=75"
            srcSet="/_next/image?url=%2Fimages%2Floveindustries.jpg&w=384&q=75 1x, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=640&q=75 2x"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </div>
        <div
          className="Showstyled__StyledMetadata-sc-7uvw6o-2 hYvsDa"
        >
          <div
            className="Showstyled__StyledEpisodes-sc-7uvw6o-3 kBHXUu"
          >
            21
             episodes
          </div>
          <h3
            className="Showstyled__StyledTitle-sc-7uvw6o-4 ddSRVy"
          >
            Love Industries
          </h3>
        </div>
      </div>
    </div>
  </div>
  <div
    className="ShowOverviewstyled__StyledShowSelectorWrapper-xnlc2h-1 iGydrH"
  >
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        1
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector Gaycation"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fgaycation.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fgaycation.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fgaycation.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        2
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector Huang's World"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fhuangsworld.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        3
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector King of The Road"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fkingoftheroad.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        4
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector Last Chance High"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Flastchancehigh.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 jBVUas"
      >
        5
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector Love Industries"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 gfvZOF"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Floveindustries.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Floveindustries.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Floveindustries.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        6
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector pilotweek"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fpilotweek.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fpilotweek.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fpilotweek.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        7
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector States of Undress"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fstatesofundress.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        8
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector VICE"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fvicehbo.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fvicehbo.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fvicehbo.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        9
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector Woman"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fwoman.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fwoman.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fwoman.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="ShowSelectorstyled__StyledWrapper-sc-1mhcrp9-0 ihnzvx"
    >
      <div
        className="ShowSelectorstyled__StyledShowLabel-sc-1mhcrp9-3 bvPzBU"
      >
        10
      </div>
      <div
        style={Object {}}
      >
        <div
          className="ShowSelectorstyled__StyledImageWrapper-sc-1mhcrp9-2 jYOuUx"
        >
          <div
            style={
              Object {
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "left": 0,
                "margin": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <img
              alt="show selector VICE World of Sports"
              className="ShowSelectorstyled__StyledShowSelector-sc-1mhcrp9-1 fEgwBO"
              decoding="async"
              onClick={[Function]}
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Fworldofsports.jpg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimages%2Fworldofsports.jpg&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fworldofsports.jpg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
